packages/services/src/linkResolver.service.ts

```diff
@@ -351,6 +351,15 @@ export const buildElementString = (ai: any) => {
 };
 
 export const extractFromElementString = (elementString: string) => {
+  let preparedElementString = elementString;
+  if (!preparedElementString) {
+    return {};
+  }
+
+  if (!preparedElementString.startsWith('01') && !preparedElementString.startsWith('(01)')) {
+    preparedElementString = `(01)${preparedElementString}`;
+  }
+
   const gs1DigitalLinkToolkit = new GS1DigitalLinkToolkit();
-  return gs1DigitalLinkToolkit.extractFromGS1elementStrings(elementString);
+  return gs1DigitalLinkToolkit.extractFromGS1elementStrings(preparedElementString);
```

Comment:
I have a question is if the `elementString` input parameter is '054123450000133103000189', will this function run successfully, or does the `extractFromElementString` function only process one AI at a time?

The example provided `elementString` combines two element strings (2 AI):
AI: 01 (by default), value: 05412345000013
AI: 3103, value: 000189

Reply:
Yes, this code makes sure the `elementString` must have the AI code 01